database/instance_db.go

```diff
@@ -120,7 +126,31 @@ func (d *LevelDBInstanceDB) Close() error {
 	return d.db.Close()
 }
 
-// Delete deletes service from database.
+// Delete deletes an instance from database.
 func (d *LevelDBInstanceDB) Delete(hash hash.Hash) error {
 	return d.db.Delete(hash, nil)
 }
+
+// ErrInstanceNotFound is an not found error.
+type ErrInstanceNotFound struct {
```

Comment:
What is the reason for splitting this and extending api? In that case we end up with err not found for every type.

can't we merge those errors into one like:

```go
type ErrNotFound struct {
  hash hash.Hash
  label string
}
```

func (e *ErrNotFound) Error() string {
  return fmt.Sprintf("database: %s %q not found", e.label, e.hash)
}
```

Reply:
That's what I start doing it first but then I revert to create new types..

Reply:
not sure we even need a label, we should have it based on the context and the api we call

Reply:
the context of the apu is not enough, a call to instance APIs can fetch a service and thus return an instance error (because of context) instead of a service.